ambry-store/src/main/java/com/github/ambry/store/BlobStoreStats.java

```diff
@@ -202,6 +208,37 @@
     return deleteTombstoneStats;
   }
 
+  /**
+   * Callback method when a cycle of compaction is finished.
+   */
+  public void onCompactionFinished() {
+    if (bucketCount > 0) {
+      // Compaction is finished, we need to reconstruct the log segments and physical storage usage in the scan result.
+      scanLock.lock();
+      try {
+        long delay = indexScannerScheduledFuture.getDelay(TimeUnit.SECONDS);
+        if (delay > 0) {
+          // We haven't started the next index scanner tasks, just cancel it and reschedule.
+          indexScannerScheduledFuture.cancel(false);
```

Comment:
What's the reason for rescheduling the task to a later time in this case instead of just letting it run (since it has not started running yet)?